src/MusicTheory/PitchClass/Enharmonic.elm

```diff
@@ -0,0 +1,245 @@
+        ( Internal.letter pitchClass, Natural )
+
+    else if Internal.offset pitchClass < 0 then
+        naturalOrSingleFlat pitchClass
+
+    else
+        naturalOrSingleSharp pitchClass
+
+
+simplifiedToString : PitchClass -> String
+simplifiedToString pitchClass =
+    simplified pitchClass |> toString
+
+
+{-| Returns the enharmonic equivalent pitch class expressed as a note from the diatonic C major scale that is natural or lowered once.
+
+    naturalOrSingleFlat (pitchClass F DoubleSharp) == ( G, Natural )
+
+    naturalOrSingleFlat (pitchClass C TripleSharp) == ( E, Flat )
+
+-}
+naturalOrSingleFlat : PitchClass -> ( Letter, Accidental )
+naturalOrSingleFlat pitchClass =
+    case pitchClass |> PitchClass.semitones |> semitonesToNaturalOrAccidental of
+        Nat letter ->
+            ( letter, Natural )
+
+        SharpFlat _ letter ->
+            ( letter, Flat )
+
+
+{-| Returns the enharmonic equivalent pitch class expressed as a note from the diatonic C major scale that is natural or raised once
+
+    naturalOrSingleSharp (pitchClass F DoubleSharp) == ( G, Natural )
+
+    naturalOrSingleSharp (pitchClass C TripleSharp) == ( D, Sharp )
+
+-}
+naturalOrSingleSharp : PitchClass -> ( Letter, Accidental )
+naturalOrSingleSharp pitchClass =
+    case pitchClass |> PitchClass.semitones |> semitonesToNaturalOrAccidental of
+        Nat letter ->
+            ( letter, Natural )
+
+        SharpFlat letter _ ->
+            ( letter, Sharp )
+
+
+{-| String representation of a letter and an accidental.
+
+    toString ( D, Sharp ) == "D♯"
+
+-}
+toString : ( Letter, Accidental ) -> String
+toString ( letter, accidental ) =
+    accidentalToString accidental ++ letterToString letter
+
+
+
+-- INTERNALS
+
+
+letterToString : Letter -> String
+letterToString letter =
+    case letter of
+        C ->
+            "C"
+
+        D ->
+            "D"
+
+        E ->
+            "E"
+
+        F ->
+            "F"
+
+        G ->
+            "G"
+
+        A ->
+            "A"
+
+        B ->
+            "B"
+
+
+accidentalToString : Accidental -> String
+accidentalToString accidental =
+    case accidental of
+        TripleFlat ->
+            "♭𝄫"
+
+        DoubleFlat ->
+            "𝄫"
+
+        Flat ->
+            "♭"
+
+        Natural ->
+            ""
+
+        Sharp ->
+            "♯"
+
+        DoubleSharp ->
+            "𝄪"
+
+        TripleSharp ->
+            "♯𝄪"
+
+
+offsetToAccidental : Internal.Offset -> Maybe Accidental
+offsetToAccidental (Internal.Offset offset) =
+    if offset == -3 then
+        Just TripleFlat
+
+    else if offset == -2 then
+        Just DoubleFlat
+
+    else if offset == -1 then
+        Just Flat
+
+    else if offset == 0 then
+        Just Natural
+
+    else if offset == 1 then
+        Just Sharp
+
+    else if offset == 2 then
+        Just DoubleSharp
+
+    else if offset == 3 then
+        Just TripleSharp
+
+    else
+        Nothing
+
+
+type NaturalOrAccidental
```

Comment:
I don't see why we need this type. We could just write functions:
```
enharmonicEquivalents : PitchClass -> Int -> List PitchClass
enharmonicEquivalents thePitchClass maxAccidentals =
```

That returned all enharmonic equivalents of the `thePitchClass` with accidentals from 0 to `maxAccidentals`.

Reply:
This is a private function that is necessary for the other calculations. What you describe makes sense but I see that as an addition rather than a replacement.